modules/storage-api/src/main/java/org/apache/ignite/internal/storage/engine/StorageEngine.java

```diff
@@ -57,4 +57,12 @@ public interface StorageEngine {
      */
     // TODO: IGNITE-19717 Get rid of indexDescriptorSupplier
     MvTableStorage createMvTable(StorageTableDescriptor tableDescriptor, StorageIndexDescriptorSupplier indexDescriptorSupplier);
+
+    /**
+     * Destroys the table on node recovery if it exists.
+     *
+     * @param tableId Table ID.
+     * @throws StorageException If an error has occurs while dropping the table.
+     */
+    void dropMvTableOnRecovery(int tableId);
```

Comment:
I think this method's name is too specific. I would suggest to call it `dropMvTable` or `destroyMvTable`

Reply:
Why not.